tests/sanity/test_cilium_rock.py

```diff
@@ -0,0 +1,72 @@
+#
+# Copyright 2025 Canonical, Ltd.
+#
+
+import shlex
+
+import pytest
+from k8s_test_harness.util import docker_util, env_util, fips_util
+from test_util import config, rock
+
+IMAGE_NAME = "cilium"
+IMAGE_BASE = f"ghcr.io/canonical/{IMAGE_NAME}"
+IMAGE_ENTRYPOINT = "cilium-agent --version"
+
+
+@pytest.fixture
+def skip_if_nonfips(GOFIPS, image_version):
+    rockcraft_yaml = (
+        (config.REPO_PATH / image_version / "cilium" / "rockcraft.yaml")
+        .read_text()
+        .lower()
+    )
+
+    if GOFIPS == 1 and not fips_util.is_fips_rock(rockcraft_yaml):
+        pytest.skip("Skipping because this version of rock is not built for FIPS")
+
+
+@pytest.mark.parametrize(
+    "image_version", env_util.image_versions_in_repo(IMAGE_NAME, config.REPO_PATH)
+)
+def test_executable(image_version):
+    rock.run_image(IMAGE_NAME, image_version, IMAGE_ENTRYPOINT, config.IMAGE_ARCH)
+
+
+@pytest.mark.parametrize(
+    "image_version", env_util.image_versions_in_repo(IMAGE_NAME, config.REPO_PATH)
+)
+def test_pebble_executable(image_version):
+    rock.check_pebble(IMAGE_NAME, image_version, "v1.18.0", config.IMAGE_ARCH)
+
+
+@pytest.mark.parametrize("GOFIPS", [0, 1], ids=lambda v: f"GOFIPS={v}")
+@pytest.mark.parametrize(
+    "image_version", env_util.image_versions_in_repo(IMAGE_NAME, config.REPO_PATH)
+)
+def test_fips(skip_if_nonfips, image_version, GOFIPS):
+    image = env_util.get_build_meta_info_for_rock_version(
+        IMAGE_NAME, image_version, config.IMAGE_ARCH
+    ).image
+    entrypoint = shlex.split(IMAGE_ENTRYPOINT)
+
+    rockcraft_yaml = (
+        (config.REPO_PATH / image_version / "cilium" / "rockcraft.yaml")
+        .read_text()
+        .lower()
+    )
```

Comment:
Then you can use the function here and properly decide if the runner should skip the test.
```suggestion
@pytest.mark.parametrize("gofips", [0, 1], ids=lambda v: f"GOFIPS={v}")
@pytest.mark.parametrize(
    "image_version", env_util.image_versions_in_repo(IMAGE_NAME, config.REPO_PATH)
)
def test_fips(image_version: str, gofips: int):
    rockcraft_yaml = get_rockcraft_yaml(image_version)
    
    if gofips == 1 and not fips_util.is_fips_rock(rockcraft_yaml):
        pytest.skip("This version of the ROCK is not built for FIPS")
    
    image = env_util.get_build_meta_info_for_rock_version(
        IMAGE_NAME, image_version, config.IMAGE_ARCH
    ).image
    entrypoint = shlex.split(IMAGE_ENTRYPOINT)
```